src/managers/PointerManager.cpp

```diff
@@ -668,30 +668,46 @@ void CPointerManager::move(const Vector2D& deltaLogical) {
 }
 
 void CPointerManager::warpAbsolute(Vector2D abs, SP<IHID> dev) {
-
-    PHLMONITOR currentMonitor = g_pCompositor->m_pLastMonitor.lock();
-    if (!currentMonitor || !dev)
+    if (!dev)
         return;
 
     if (!std::isnan(abs.x))
         abs.x = std::clamp(abs.x, 0.0, 1.0);
     if (!std::isnan(abs.y))
         abs.y = std::clamp(abs.y, 0.0, 1.0);
 
-    // in logical global
-    CBox mappedArea = currentMonitor->logicalBox();
+    // find x and y size of the entire space
+    Vector2D bottomRight = {-9999999, -9999999}, topLeft = {9999999, 9999999};
```

Comment:
g_pCompositor->m_vMonitors.at(0).vecPosition, g_pCompositor->m_vMonitors.at(0).vecPosition + g_pCompositor->m_vMonitors.at(0).vecSize

then skip iterating the first one